### PR TITLE
fixes internal error: no generic body fixes #1500

### LINKED ISSUE #1500: internal error: no generic body
### Example

```nim
type
  TFtpBase*[SockType] = object
    job: TFTPJob[SockType]

  PFtpBase*[SockType] = ref TFtpBase[SockType]
  TFtpClient* = TFtpBase[string]
  TFTPJob[T] = object
```

### Current Output

```
Error: internal error: no generic body
```

### Expected Output

This should compile

### Workaround

change the order of declarations

```nim
type
  TFTPJob[T] = object
  TFtpBase*[SockType] = object
    job: TFTPJob[SockType]

  PFtpBase*[SockType] = ref TFtpBase[SockType]
  TFtpClient = TFtpBase[string]
```

### Additional Information

```
$ nim -v
Nim Compiler Version 0.19.9 [Linux: amd64]
Compiled at 2019-04-26
Copyright (c) 2006-2019 by Andreas Rumpf

git hash: e48f420ab0161d6fc46c72a5a4f0142e77ed9796
active boot switches: -d:release
```

### PR DESCRIPTION
**Pull request by jmgomez**
fixes internal error: no generic body fixes #1500

### CODE REVIEW

**Comment by jmgomez** [comment]
Just noticed it also fixes #14660 which is a duplicated

**Review by jmgomez — COMMENTED** [review]
I dont understand the change. What's your intention? By the time it gets into the var declaration the type section is already sem

**Comment by bung87** [comment]
> I dont understand the change. What's your intention? By the time it gets into the var declaration the type section is already sem

just ensure it will not face downstream error BTW it lose last empty line.

File: compiler/semtypes.nim [hunk 1]
```diff
@@ -1578,6 +1578,13 @@ proc semGeneric(c: PContext, n: PNode, s: PSym, prev: PType): PType =
         localError(c.config, n.info, errCannotInstantiateX % s.name.s)
         result = newOrPrevType(tyError, prev, c)
       else:
+        #fixes 1500
+        if n[0].kind == nkSym and n[0].sym.ast != nil and n[0].sym.ast.len > 1 and n[0].sym.ast[2].kind == nkObjectTy:
```
Comment (Araq) [inline compiler/semtypes.nim]: Turn this whole check into a helper proc please. Name it `containsGenericInvocationWithForward`

**Pull request merged**